src/hostname/hostnamed.c

```diff
@@ -300,7 +301,24 @@ static const char* fallback_chassis(void) {
                 log_debug("Unhandled ACPI PM profile 0x%02x, ignoring.", t);
         }
 
-        return NULL;
+try_devicetree:
+        r = read_one_line_file("/sys/firmware/devicetree/base/chassis-type", &type);
```

Comment:
I know this is already merged but this sounds relevant for here?

> Userspace must not use the /sys/firmware/devicetree/base
> 		path directly, but instead should follow /proc/device-tree
>		symlink. It is possible that the absolute path will change
>		in the future, but the symlink is the stable ABI.

Source: https://www.kernel.org/doc/Documentation/ABI/testing/sysfs-firmware-ofw